src/orion/core/io/experiment_builder.py

```diff
@@ -204,6 +204,9 @@ def build(name, version=None, branching=None, **config):
         raise NoConfigurationError(
             'Experiment {} does not exist in DB and space was not defined.'.format(name))
 
+    if len(config['space']) == 0:
+        raise ValueError("No prior found. Please include at least one.")
```

Comment:
We would need to update the documentation here:
https://github.com/Epistimio/orion/blob/develop/src/orion/client/__init__.py#L145

Reply:
Also, ValueError is not detected by CLI but NoConfigurationError is (printing user-friendly error message vs full stack-trace). I would be in favor of merging the two are reword the error message if necessary. It's essentially the same error.

Reply:
I opted to change the exception to `NoConfigurationError`, this way the message is still precise enough when shown to the user.